server/src/main/java/org/opensearch/rest/pagination/IndexBasedPaginationStrategy.java

```diff
@@ -0,0 +1,191 @@
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.rest.pagination;
+
+import org.opensearch.OpenSearchParseException;
+import org.opensearch.cluster.ClusterState;
+import org.opensearch.cluster.metadata.IndexMetadata;
+import org.opensearch.common.Nullable;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import static org.opensearch.rest.pagination.PaginatedQueryRequest.PAGINATED_QUERY_ASCENDING_SORT;
+
+/**
+ * This strategy can be used by the Rest APIs wanting to paginate the responses based on Indices.
+ * The strategy considers create timestamps of indices as the keys to iterate over pages.
+ *
+ * @opensearch.internal
+ */
+public class IndexBasedPaginationStrategy implements PaginationStrategy<String> {
+
+    private final PaginatedQueryResponse paginatedQueryResponse;
+    private final List<String> indicesFromRequestedToken;
+
+    private static final String DEFAULT_INDICES_PAGINATED_ELEMENT = "indices";
+
+    public IndexBasedPaginationStrategy(PaginatedQueryRequest paginatedQueryRequest, ClusterState clusterState) {
+        // Get list of indices metadata sorted by their creation time and filtered by the last send index
+        List<IndexMetadata> sortedIndicesList = PaginationStrategy.getSortedIndexMetadata(
+            clusterState,
+            getMetadataListFilter(paginatedQueryRequest.getRequestedTokenStr(), paginatedQueryRequest.getSort()),
+            getMetadataListComparator(paginatedQueryRequest.getSort())
+        );
+        List<IndexMetadata> metadataListForRequestedToken = getMetadataListForRequestedToken(sortedIndicesList, paginatedQueryRequest);
+        this.indicesFromRequestedToken = metadataListForRequestedToken.stream()
+            .map(metadata -> metadata.getIndex().getName())
+            .collect(Collectors.toList());
+        this.paginatedQueryResponse = getPaginatedResponseForRequestedToken(paginatedQueryRequest.getSize(), sortedIndicesList);
+    }
+
+    private static Predicate<IndexMetadata> getMetadataListFilter(String requestedTokenStr, String sortOrder) {
+        boolean isAscendingSort = sortOrder.equals(PAGINATED_QUERY_ASCENDING_SORT);
+        IndexStrategyToken requestedToken = Objects.isNull(requestedTokenStr) || requestedTokenStr.isEmpty()
+            ? null
+            : new IndexStrategyToken(requestedTokenStr);
+        if (Objects.isNull(requestedToken)) {
+            return indexMetadata -> true;
+        }
+        return indexMetadata -> {
+            if (indexMetadata.getIndex().getName().equals(requestedToken.nameOfLastRespondedIndex)) {
+                return false;
+            } else if (indexMetadata.getCreationDate() == requestedToken.creationTimeOfLastRespondedIndex) {
```

Comment:
creationTimeOfLastRespondedIndex - LastIndexCreationTime
nameOfLastRespondedIndex. - LastIndexName

Reply:
Updated in the revision.